### PR TITLE
Add sqs error handling

### LINKED ISSUE #1799: Moto should raise an error when SQS send_message_batch pass an empty entries
When doing `client.send_message_batch(QueueUrl=queue_url, Entries=[])`, boto will raise an error.

```
Traceback (most recent call last):
  File "<stdin>", line 3, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/botocore/client.py", line 314, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/botocore/client.py", line 612, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.errorfactory.EmptyBatchRequest: An error occurred (AWS.SimpleQueueService.EmptyBatchRequest) when calling the SendMessageBatch operation: There should be at least one SendMessageBatchRequestEntry in the request.
```

moto should raise the same error, rather than to be silent on that.

Environment:

```
$ python --version
Python 2.7.11
$ python -c "import boto3; print(boto3.__version__)"
1.4.7
$ python -c "import moto; print(moto.__version__)"
1.3.1
```

### PR DESCRIPTION
**Pull request by gruebel**
I had to rewrite `sqs.send_message_batch` a bit to be able to handle most errors.

Also fixes #1799

Over time I will also check other functions in the SQS implementation and update the coverage file while doing so.

### CODE REVIEW

File: moto/sqs/responses.py [hunk 1]
```diff
@@ -397,6 +439,20 @@ def tag_queue(self):
         queue_name = self._get_queue_name()
         tags = self._get_map_prefix('Tag', key_end='.Key', value_end='.Value')
 
+        try:
+            self.sqs_backend.get_queue(queue_name)
+        except QueueDoesNotExist as e:
+            return self._error('AWS.SimpleQueueService.NonExistentQueue',
+                               e.description)
+
+        if len(tags) == 0:
```
Comment (mikegrima) [inline moto/sqs/responses.py]: Make this `if not len(tags):`

File: moto/sqs/responses.py [hunk 2]
```diff
@@ -242,25 +242,59 @@ def send_message_batch(self):
 
         queue_name = self._get_queue_name()
 
+        try:
+            self.sqs_backend.get_queue(queue_name)
+        except QueueDoesNotExist as e:
```
Comment (mikegrima) [inline moto/sqs/responses.py]: Is this needed? If it raises a `QueueDoesNotExist`, that should have all the info required to format the exception to the client.

File: moto/sqs/responses.py [hunk 3]
```diff
@@ -397,6 +439,20 @@ def tag_queue(self):
         queue_name = self._get_queue_name()
         tags = self._get_map_prefix('Tag', key_end='.Key', value_end='.Value')
 
+        try:
+            self.sqs_backend.get_queue(queue_name)
+        except QueueDoesNotExist as e:
```
Comment (mikegrima) [inline moto/sqs/responses.py]: Ditto on if this is needed.

File: moto/sqs/responses.py [hunk 4]
```diff
@@ -418,7 +474,11 @@ def untag_queue(self):
     def list_queue_tags(self):
         queue_name = self._get_queue_name()
 
-        queue = self.sqs_backend.get_queue(queue_name)
+        try:
+            queue = self.sqs_backend.get_queue(queue_name)
+        except QueueDoesNotExist as e:
```
Comment (mikegrima) [inline moto/sqs/responses.py]: Ditto.

File: moto/sqs/responses.py (hunk 2 above)
Comment (gruebel) [reply moto/sqs/responses.py]: The problem is the general implementation of the `QueueDoesNotExist` exception. It results in an 503 error while running the moto in server mode. I would need to change it to a `RESTError` then the handling would be right, I think.

File: moto/sqs/responses.py (hunk 1 above)
Comment (gruebel) [reply moto/sqs/responses.py]: Done

File: moto/sqs/responses.py (hunk 2 above)
Comment (mikegrima) [reply moto/sqs/responses.py]: Thank You -- that makes sense.

**Pull request merged**